lib/src/native/realm_core.dart

```diff
@@ -1787,19 +1802,63 @@ final _nativeFinalizer = NativeFinalizer(_realmLib.addresses.realm_release);
 
 abstract class HandleBase<T extends NativeType> implements Finalizable {
   final Pointer<T> _pointer;
+  bool released = false;
+  final bool isUnowned;
 
   @pragma('vm:never-inline')
   void keepAlive() {}
 
-  HandleBase(this._pointer, int size) {
+  HandleBase(this._pointer, int size) : isUnowned = false {
     _nativeFinalizer.attach(this, _pointer.cast(), detach: this, externalSize: size);
-    if (_enableFinalizerTrace) _setupFinalizationTrace(this, _pointer);
+    if (_enableFinalizerTrace) {
+      _setupFinalizationTrace(this, _pointer);
+    }
   }
 
-  HandleBase.unowned(this._pointer);
+  HandleBase.unowned(this._pointer) : isUnowned = true;
 
   @override
   String toString() => "${_pointer.toString()} value=${_pointer.cast<IntPtr>().value}";
+
+  /// @nodoc
+  /// A method that will be invoked just before the handle is released. Allows to cleanup
+  /// any custom data that inheritors are storing.
+  void _releaseCore() {}
+
+  void release() {
+    if (released) {
+      return;
+    }
+
+    _releaseCore();
```

Comment:
We don't need to run prior to the native finalizer. This is only intended as a cleanup step to prevent the list of children from growing endlessly. I added a managed finalizer based on your suggestion - LMK if that's what you had in mind.